easybuild/tools/systemtools.py

```diff
@@ -210,15 +291,37 @@ def get_cpu_model():
     os_type = get_os_type()
 
     if os_type == LINUX and os.path.exists(PROC_CPUINFO_FP):
-        # we need 'model name' on Linux/x86, but 'model' is there first with different info
-        # 'model name' is not there for Linux/POWER, but 'model' has the right info
-        model_regex = re.compile(r"^model(?:\s+name)?\s+:\s*(?P<model>.*[A-Za-z].+)\s*$", re.M)
         txt = read_file(PROC_CPUINFO_FP)
-        res = model_regex.search(txt)
-        if res is not None:
-            model = res.group('model').strip()
-            _log.debug("Determined CPU model on Linux using regex '%s' in %s: %s",
-                       model_regex.pattern, PROC_CPUINFO_FP, model)
+
+        arch = get_cpu_architecture()
+        if arch in [AARCH32, AARCH64]:
+            # On ARM platforms, no model name is provided in /proc/cpuinfo.  However, for vanilla ARM cores
+            # we can reverse-map the part number.
+            vendor = get_cpu_vendor()
+            if vendor == ARM:
+                model_regex = re.compile(r"CPU part\s+:\s*(\S+)", re.M)
+                # There can be big.LITTLE setups with different types of cores!
+                res = model_regex.findall(txt)
+                if res:
+                    model_ids = list(set(res))
+                    id_list = []
+                    for model_id in model_ids:
```

Comment:
Will the iteration order be the same for every run?  OK, I may have missed a `sort()` call in my first attempt, but that's a different story...